src/Bicep.Core/Semantics/Namespaces/AzNamespaceSymbol.cs

```diff
@@ -327,8 +329,105 @@ private static IEnumerable<FunctionOverload> GetAzOverloads(ResourceScope resour
                 .Build();
         }
 
+        private static IEnumerable<Decorator> GetAzDecorators()

```

Comment:
Was there any discussion on whether decorators should be namespaced, and if so, whether they should belong to az or sys, and what the semantics should be?

Reply:
I think decorators should live in some namespace. We may allow adding new decorators via extensions in the future and this will prevent clashes.

We should put these in the `sys` namespace. `az` is more intended for Azure resource specific functionality where `sys` is for core runtime/language capabilities.

---
In reply to: [563791970](https://github.com/Azure/bicep/pull/1324#discussion_r563791970) [](ancestors = 563791970)

Reply:
Makes sense. Moved the decorators to `sys`.